File: samples/03_kmp-compose-ui/androidApp/src/main/kotlin/com/example/kmpsamples/kmpcomposeui/android/HomeViewModel.kt
package com.example.kmpsamples.kmpcomposeui.android

import com.example.kmpsamples.kmpcomposeui.shared.HomeUiState

class HomeViewModel(
    private val repository: TaskRepository = TaskRepository(TaskDataSource())
) {
    fun load(platform: String): HomeUiState = HomeUiState(
        title = "Native HomeViewModel on $platform",
        body = repository.loadTaskSummary(platform)
    )

    fun refresh(platform: String): HomeUiState = HomeUiState(
        title = "Native HomeViewModel refreshed on $platform",
        body = repository.loadTaskSummary(platform)
    )
}

File: samples/03_kmp-compose-ui/iosApp/Sources/HomeViewModel.swift
import Shared

final class HomeViewModel {
    private let repository: TaskRepository

    init(repository: TaskRepository = TaskRepository(dataSource: TaskDataSource())) {
        self.repository = repository
    }

    func load(platform: String) -> HomeUiState {
        HomeUiState(title: "Native HomeViewModel on \(platform)", body: repository.loadTaskSummary(platform: platform))
    }

    func refresh(platform: String) -> HomeUiState {
        HomeUiState(title: "Native HomeViewModel refreshed on \(platform)", body: repository.loadTaskSummary(platform: platform))
    }
}
